nestkernel/event.h

```diff
@@ -742,73 +738,82 @@ class SecondaryEvent : public Event
 
   //! size of event in units of uint_t
   virtual size_t size() = 0;
-  virtual fwit& operator<<( fwit& pos ) = 0;
-  virtual fwit& operator>>( fwit& pos ) = 0;
+  virtual std::vector< uint_t >::iterator& operator<<( std::vector< uint_t >::iterator& pos ) = 0;
+  virtual std::vector< uint_t >::iterator& operator>>( std::vector< uint_t >::iterator& pos ) = 0;
 };
 
 /**
- * Iterator class for the GapJEvent data array.
- * Generating a copy of the array at several times is too time consuming,
- * therefore we use iterators pointing to the "old" memory location of the array
+ * This template function returns the number of uints covered by a variable of
+ * type T. This function is used to determine the storage demands for a variable of
+ * type T in the NEST communication buffer, which is of type std::vector< uint_t>.
  */
-class CoeffArrayIterator
+template < typename T >
+size_t
+number_of_uints_covered( void )
 {
-
-  friend class GapJEvent;
-
-private:
-  fwit pos_;
-
-public:
-  CoeffArrayIterator()
+  size_t num_uints = sizeof( T ) / sizeof( uint_t );
+  if ( num_uints * sizeof( uint_t ) < sizeof( T ) )
   {
+    num_uints += 1;
   }
+  return num_uints;
+}
 
-  CoeffArrayIterator( const fwit& rhs )
-    : pos_( rhs )
-  {
-  }
+/**
+ * This template function writes data of type T to a given position of a
+ * std::vector< uint_t >. Please note that this function does not increase the
+ * size of the vector, it just writes the data to the position given by the iterator.
+ * The function is used to write data from SecondaryEvents to the NEST communcation buffer.
+ */
+template < typename T >
+void
+write_to_comm_buffer( T d, std::vector< uint_t >::iterator& pos )
+{
+  char* c = reinterpret_cast< char* >( &d );
 
-  CoeffArrayIterator( const dfwit& rhs )
-  {
-    fwit iit( reinterpret_cast< uint_t* >( &( *rhs ) ) );
-    pos_ = iit;
-  }
+  size_t num_uints = number_of_uints_covered< T >();
+  size_t left_to_copy = sizeof( T );
 
-  CoeffArrayIterator( const CoeffArrayIterator& rhs )
-    : pos_( rhs.pos_ )
+  for ( int i = 0; i < num_uints; i++ )
   {
+    memcpy( &( *( pos + i ) ),
+      c + i * sizeof( uint_t ),
+      left_to_copy < sizeof( uint_t ) ? left_to_copy : sizeof( uint_t ) );
```

Comment:
`std::min(left_to_copy, sizeof( uint_t ))` would be shorter and more expressive.